### PR TITLE
General Plugin Improvements

### PR DESCRIPTION
**Pull request by joemcgill**
- [x] Fix docs typos
- [x] Fix ESLint errors
- [x] Add escaping
- [x] Fix asset loading during dev
- [x] Fix PHP style

**Reviewer Note:** I didn't add CS rules to the plugin, but fixed ESLint based on HM standards and PHPCS based on WordPress-VIP standards. On the latter, I've intentionally ignored the `curl()` rule for now because it's only being used to check for API status, and switching to `wp_remote_get()` is changing the expected result.

### CODE REVIEW

File: embed-gutenberg-block-google-maps.php [hunk 1]
```diff
@@ -14,168 +14,171 @@
...
+	// URL encode the location for Google Maps.
+	$location = rawurlencode( $location );
+
+	// Set the API URL based to embed or static maps based on the interactive setting.
+	$api_url = ( $interactive ) ? "https://www.google.com/maps/embed/v1/place?key=${api_key}&q=${location}&zoom=${zoom}&maptype=${mapType}" : "https://maps.googleapis.com/maps/api/staticmap?center=${location}&zoom=${zoom}&size=${maxWidth}x${maxHeight}&maptype=${mapType}&key=${api_key}"; /* phpcs:ignore */
+
+	// Check status code of the API.
+	// phpcs:disable WordPress.WP.AlternativeFunctions
+	$ch = curl_init( $api_url );
+	curl_setopt( $ch, CURLOPT_HEADER, true );
```
Comment (peterwilsoncc) [inline embed-gutenberg-block-google-maps.php:104]: Looks like this is the option that prevents the use of `WP_HTTP`, the remaining options are defaults/can be set via the API.

Part interesting fact/part desire to check if it's essential.

File: embed-gutenberg-block-google-maps.php [hunk 2]
```diff
@@ -14,168 +14,171 @@
...
+	curl_close( $ch );
+	// phpcs:enable WordPress.WP.AlternativeFunctions
+
+	// Don't output anything if the response from Google Maps isn't a 200.
+	if ( 200 !== $httpcode ) {
+		return;
+	}
+
+	// Set the appropriate CSS class names.
+	$classes = ( $interactive ) ? "wp-block-pantheon-google-map interactive ratio$aspectRatio" : 'wp-block-pantheon-google-map'; /* phpcs:ignore */
```
Comment (peterwilsoncc) [inline embed-gutenberg-block-google-maps.php]: `/* phpcs:ignore */` ignores two lines, would be good to put it on line 117.5

Non-blocker

File: embed-gutenberg-block-google-maps.php [hunk 3]
```diff
@@ -14,168 +14,171 @@
...
+
+	// Create the output.
+	$output = "<div class='$classes'><div class='map'>";
```
Comment (peterwilsoncc) [inline embed-gutenberg-block-google-maps.php]: We'll need to add a couple of instances of escaping around here:

* `esc_attr( $classes )`
* `esc_url( $api_url )` x2

**Review by peterwilsoncc — APPROVED** [review]
This looks good generally, the only real blocker is the lack of escaping of variables in the output. Once that is fixed I think it's good to go.

Reviewed with whitespace suppressed.

File: embed-gutenberg-block-google-maps.php (hunk 3 above)
Comment (joemcgill) [reply embed-gutenberg-block-google-maps.php]: Oh weird. Was handled in 0a25177c but must have gotten lost in another commit. Will redo.

**Comment by joemcgill** [comment]
@peterwilsoncc I've fixed the critical issues and learned a new thing about PHPCS ignore rules. Cheers.

File: embed-gutenberg-block-google-maps.php (hunk 1 above)
Comment (rmccue) [reply embed-gutenberg-block-google-maps.php:104]: @peterwilsoncc It's never actually used, so irrelevant. You could replace this code with:

```php
$response = wp_remote_head( $api_url );
if ( is_wp_error( $response ) || wp_remote_retrieve_response_code( $response ) !== 200 ) {
    return;
}
```

**Pull request merged**